### PR TITLE
Fix: Fix dashboards not showing in folders with search v2 enabled

### PR DESCRIPTION
**Pull request by joshhunt**
Fixes regression introduced in https://github.com/grafana/grafana/pull/68617 where dashboards would not show for folders when Search v2 was enabled.

The issue here was converting `page` based pagination paraemter to `from`, 

 - Folders API and Search v1 uses 1-indexed pages for pagination - the first page is `?page=1`
 - Search v2 uses `from` parameter to indicate how many results to skip. First page is `?from=0&limit=50`, second page is `?from=50&limit=50
 - The frontend abstraction over both search methods uses `from`-based pagination.
 - `listDashboards` accepts a `page`, and converts that to `limit` for the searcher, but it did this incorrectly - it simply did `from = page * pageSize` which will attempt to load the first page by skipping the first 50 results~
 - The SearchV1/SQLSearcher frontend also contains an incorrect conversion of limit to page which _undoes_ the previous bug, so it was never noticed with search v1
    - Note: this is doubly on me - I briefly noticed that this conversion seemed wrong, but everything worked as expected so I carried on

Note - I will add tests to this before merging.

### CODE REVIEW

File: public/app/features/search/service/sql.ts [hunk 1]
```diff
@@ -81,7 +81,10 @@ export class SQLSearcher implements GrafanaSearcher {
     }
 
     const limit = query.limit ?? (query.from !== undefined ? 1 : DEFAULT_MAX_VALUES);
-    const page = query.from !== undefined ? query.from / limit : undefined;
+    const page =
+      query.from !== undefined
+        ? query.from / limit + 1 // pages are 1-indexed
+        : undefined;
```
Comment (ashharrison90) [inline public/app/features/search/service/sql.ts:87]:
```suggestion
    const page =
      query.from !== undefined
        ? (query.from / limit) + 1 // pages are 1-indexed
        : undefined;
```

can we bracket this or does prettier just undo it? reads much cleaner with the brackets imo

Comment (joshhunt) [reply public/app/features/search/service/sql.ts:87]: I agree, but prettier undoes it. I'll try a prettier-ignore comment

Comment (joshhunt) [reply public/app/features/search/service/sql.ts:87]: Looking again, I still don't think this math is correct 🤦🏻

**Review by ashharrison90 — APPROVED** [review]
nice tests 🥳

**Pull request merged**